File: 4101-maximum-product-of-three-elements-after-one-replacement/4101-maximum-product-of-three-elements-after-one-replacement.cs
public class Solution {
    public long MaxProduct(int[] nums) {
        int n = nums.Length;
        int cnt = 0;
        long ans = 1_00_000;
        long f = -ans, s = -ans;
        for (int r = 0; r < n; ++r) {
            if (nums[r] != 0)
                cnt++;
            if (Math.Abs(nums[r]) >= f) {
                s = f;
                f = Math.Abs(nums[r]);
            }
            else if (Math.Abs(nums[r]) > s && Math.Abs(nums[r]) != f)
                s = Math.Abs(nums[r]);
        }
        if (cnt <= n / 3)
            return 0;

        return Math.Abs(f * s * ans);
    }
}

File: 4101-maximum-product-of-three-elements-after-one-replacement/4101-maximum-product-of-three-elements-after-one-replacement.java
class Solution {
    public long maxProduct(int[] nums) {
        int n = nums.length;
        int cnt = 0;
        long ans = 1_00_000;
        long f = -ans, s = -ans;
        for (int r = 0; r < n; ++r) {
            if (nums[r] != 0)
                cnt++;
            if (Math.abs(nums[r]) >= f) {
                s = f;
                f = Math.abs(nums[r]);
            }
            else if (Math.abs(nums[r]) > s && Math.abs(nums[r]) != f)
                s = Math.abs(nums[r]);
        }
        if (cnt <= n / 3)
            return 0;

        return Math.abs(f * s * ans);
    }
}